packages/shared/src/models/appDataEvaluator/appDataEvaluator.ts

```diff
@@ -97,13 +102,30 @@ export class AppDataEvaluator {
    * "Hello Ada"
    */
   private evaluateExpression(expression: string) {
-    try {
-      const jsDelimited = addJSDelimeters(expression, Object.keys(this.context));
-      const evaluated = this.jsEvaluator.evaluate(jsDelimited, this.context);
-      return evaluated;
-    } catch (error) {
-      const parsed = this.templatedData.parse(expression);
-      return parsed;
+    if (this.shouldEvaluateJS(expression)) {
+      try {
+        const jsDelimited = addJSDelimiters(expression, Object.keys(this.context));
+        const evaluated = this.jsEvaluator.evaluate(jsDelimited, this.context);
+        return evaluated;
+      } catch (error) {
+        // Ignore errors as may be intended as templated data (will attempt next)
+      }
+    }
+    // If error or if not evaluated in JS continue as templated data replacement
+    const parsed = this.templatedData.parse(expression);
+    return parsed;
+  }
+
+  private shouldEvaluateJS(expression: string) {
```

Comment:
Would be nice to have a similar check for all template strings eventually, removing the need to use excessive local vars to concatenate dynamic strings (I know sometimes we'll author a local var just to store a `_` character, so we can provide a string that will be correctly interpreted by the parser, but it'd be nice to be able to write `{@local.prefix}_{@local.suffix}` inline).

Hard to see how to avoid a hack such as this though, when considering the variety of authoring patterns we support, the intention of which is always implicit.

Reply:
I thought we currently could concatenate no? I thought if explicitly including the curly braces it would work.
On the plus side, now that we have a JS-first approach I think you could also do more things like `"@local.string"+"_suffix"` (haven't tested, but I'd expect either that to work or `@local.string+"_suffix"`... doubt interpolation like `${@local.string}_suffix` would but could also be worth investigating.

But yeah ultimately at some point we will have to either commit to always supporting somewhat hacky/mixed interpretation patterns or be a bit more insistent on how to indicate something is templated vs computed

Reply:
oh wait, you mean with the current templating system. Yeah agreed, I'm hoping one day to totally overhaul that and use more of the methods that we've been slowly introducing to be consistently applied across the entire app. But that is quite a task (been doing a lot of stuff linked to it with data-items), likely needing some means to have the new templating system running in parallel with old (via some sort of opt-in deployment config)